### DISCUSSION TITLE
What is the best way we can contribute to open source project?

### DISCUSSION THREAD

**Discussion by 0xsimpi** [ows-web/hacktoberfest22#45] [Q&A] [answered]
Generally during Hacktoberfest, contributers contribute to any repository and not follow the guidence of contributing to any real open source project.

Please Guide me.

**Comment by 0xsonu** [comment, marked as answer]
> Generally during Hacktoberfest, contributers contribute to any repository and not follow the guidence of contributing to any real open source project.
> 
> Please Guide me.

### Contributing to open source can be a rewarding way to learn, teach, and build experience in just about any skill you can imagine.


**Why do people contribute to open source? Plenty of reasons!**

- Improve software you rely on
Lots of open source contributors start by being users of software they contribute to. When you find a bug in an open source software you use, you may want to look at the source to see if you can patch it yourself. If that’s the case, then contributing the patch back is the best way to ensure that your friends (and yourself when you update to the next release) will be able to benefit from it.

- Improve existing skills
Whether it’s coding, user interface design, graphic design, writing, or organizing, if you’re looking for practice, there’s a task for you on an open source project.

- Meet people who are interested in similar things
Open source projects with warm, welcoming communities keep people coming back for years. Many people form lifelong friendships through their participation in open source, whether it’s running into each other at conferences or late night online chats about burritos.

- Find mentors and teach others
Working with others on a shared project means you’ll have to explain how you do things, as well as ask other people for help. The acts of learning and teaching can be a fulfilling activity for everyone involved.

- Build public artifacts that help you grow a reputation (and a career)
By definition, all of your open source work is public, which means you get free examples to take anywhere as a demonstration of what you can do.

- Learn people skills
Open source offers opportunities to practice leadership and management skills, such as resolving conflicts, organizing teams of people, and prioritizing work.

- It’s empowering to be able to make changes, even small ones
You don’t have to become a lifelong contributor to enjoy participating in open source. Have you ever seen a typo on a website, and wished someone would fix it? On an open source project, you can do just that. Open source helps people feel agency over their lives and how they experience the world, and that in itself is gratifying.
